source/RedisInside/Redis.cs

```diff
@@ -93,13 +94,19 @@ public void Dispose()
 
         private class Config : IConfig
         {
+            private static readonly ConcurrentDictionary<int, byte> usedPorts = new ConcurrentDictionary<int, byte>();

```

Comment:
I'm assuming all the code you're having concurrency issues with is within the same app domain?